compiler/src/dotty/tools/dotc/transform/ShortcutImplicits.scala

```diff
@@ -164,13 +172,31 @@ class ShortcutImplicits extends MiniPhase with IdentityDenotTransformer { thisPh
           val fwdClosure = cpy.Block(tree)(cpy.DefDef(meth)(rhs = forwarder) :: Nil, cl)
           (remappedCore, fwdClosure)
         case EmptyTree =>
-          (_ => _ => EmptyTree, EmptyTree)
+          original.resetFlag(Deferred) // original is no longer abstract
+          val tpe = original.info.finalResultType.dealias.asInstanceOf[AppliedType]
+          val methTpe = ImplicitMethodType(tpe.args.init, tpe.args.last)
+          val tparams = mdef.tparams.map(tp => ref(tp.symbol))
+          val vparams = mdef.vparamss.map(_.map(vp => ref(vp.symbol)))
+          def forwarder(args: List[Tree]) = ref(direct)
+            .appliedToTypeTrees(tparams)
+            .appliedToArgss(vparams)
+            .appliedToArgs(args)
+          val fwdClosure = Lambda(methTpe, forwarder(_))(ctx.withOwner(original))
+          (_ => _ => EmptyTree, fwdClosure)
       }
 
       val (remappedCore, fwdClosure) = splitClosure(mdef.rhs)
-      val originalDef = cpy.DefDef(mdef)(rhs = fwdClosure)
       val directDef = transformDefDef(polyDefDef(direct.asTerm, remappedCore))
-      flatTree(List(originalDef, directDef))
+      if (original.allOverriddenSymbols.exists(!_.is(Deferred))) {
+        // if original overrides something we only generate
+        // the direct method and remove original
+        original.owner.asClass.delete(original)
```

Comment:
I am a bit scared by this. Especially how it will interact with other phases in the same group such as `ExpandSAMs`